File: src/main/java/com/partymakers/shareparty/domain/entity/PartyRoom.java
package com.partymakers.shareparty.domain.entity;

import java.util.ArrayList;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

public class PartyRoom {

    @Getter
    final int id;


    @Getter
    final String  name;

    @Getter
    @Setter
    List<Friend>  friends;

    @Getter
    @Setter
    List<Expense> expenses;

    public PartyRoom(String name){
        id = 0;
        this.name = name;
        friends   = new ArrayList<>();
        expenses  = new ArrayList<>();
    }

    public PartyRoom(int id, String name){
        this.id   = id;
        this.name = name;
        friends   = new ArrayList<>();
        expenses  = new ArrayList<>();
    }
}

File: src/main/kotlin/com/partymakers/shareparty/party/domain/entity/PartyRoom.kt
package com.partymakers.shareparty.party.domain.entity

import com.partymakers.shareparty.friends.domain.entity.Friend

internal data class PartyRoom(
    val description: PartyRoomDescription,
    val friends: Set<Friend> = setOf(),
    val expenses: List<Expense> = listOf(),
) {
    constructor(name: String) : this(
        description = PartyRoomDescription(name),
        friends = setOf(),
        expenses = listOf()
    )

    constructor(id: Long, name: String, friends: Set<Friend>, expenses: List<Expense>) : this(
        description = PartyRoomDescription(id, name),
        friends = friends,
        expenses = expenses
    )

    constructor(id: Long, name: String) : this(
        description = PartyRoomDescription(id, name),
        friends = setOf(),
        expenses = listOf()
    )
} 